snapcraft/plugins/script.py

```diff
@@ -0,0 +1,106 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""The script plugin is useful for building script based parts.
+Script based projects use bash commands which drive the build.
+This plugin uses the common plugin keywords as well as those for "sources".
+For more information check the 'plugins' topic for the former and the
+'sources' topic for the latter.
+Additionally, this plugin uses the following plugin-specific keyword:
+    - script-name:
+      (string)
+      Run the given script name.
+    - stages:
+      (list of strings)
+      Run the script in the given stages. (default: ['build'])
```

Comment:
You could have a script for pull, and a different one for build. Or you could have to run two commands in one of those phases.

I'm not quite sure about this plugin, because you are trying to wrap a wild (and sometimes a little random :) thing like a custom build script into a generic workflow with fixed phases and arguments. So there will be a lot of cases that will need modifications on this plugin to work.

I think that instead of providing a plugin to run a script, we should provide better guidelines to let people make their own custom plugin that runs whatever they need to pull, and then whatever they need to build. In these past weeks I've seen all kinds of crazy things during build, I see no good way to make them behave in a similar way.

Grumpiness aside, welcome to the snapcraft! Your contribution is really appreciated. Thank you.